### DISCUSSION TITLE
Airflow 2.8.0 in docker getting gcloud permission denied

### DISCUSSION THREAD

**Discussion by johndickson2** [apache/airflow#36844] [Q&A] [answered]
When starting airflow using the official docker image, I receive the following error for both airflow-scheduler and airflow-worked:
/bin/sh: 1: gcloud: Permission denied

The dag's that I am trying to run locally include a command using gcloud bash command and the permission denied is in turn causing my code to not run the gcloud command, any help to work around this would be greatly appreciated.

**Comment by potiuk** [comment, marked as answer]
Well that's something your custom image must add wrongly (or you did not add it a all). Airflow's reference image does not have gcloud CLI installed and you need to extend the image to add it - https://airflow.apache.org/docs/docker-stack/build.html#examples-of-image-extending and how you added the gcloud tool - is up to your extension. Maybe we can help you if you explain how you've added it, but generally you need to make sure the binary you added is accessible for the user that you use to run airflow with, and with executable permission for that user. Those are all generic things that you need to make sure when you build your own image, but not knowing how you've done it makes it rather difficult to help you.

**Reply by johndickson2** [reply]
Thank you for the help! I'm fairly new to both docker and airflow, so I was just running docker compose on the official image. I've added a Docker file and extended the original image, per your suggestions, and looks like I am on track to getting my dag working.
